engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
On line 1014 the root cause of NPE was Container.getNodeId() returning null so I think it is the same issue. My comment was related to the timing - when the AllocatedContainer object is created it is passed a Container object and I was wondering if calling getNodeId() on that (at this point in time) could return null as well.